src/redis.store.ts

```diff
@@ -76,10 +77,33 @@ export class RedisStore implements IRedisStore {
     });
   }
 
+  getItemByIndexes(channel: string, indexes: number[], callback: (e?: Error | null, results?: any) => void) {
+    const retriveOneKey = (index, cb) => {
+      this.redis.zrangebyscore(key(channel, KEYS), index, index, cb);
+    };
+
+    const retrieveKeys = (cb: (e: Error | null | undefined, res?: any | null) => void) => {
+      let tasks: any[] = [];
+      indexes.forEach(index => {
+        tasks.push(callack => retriveOneKey(index, callack));
+      });
+
+      async.series(tasks, (err, data) => {
+        if (err) {
+          return cb(err, null);
+        }
+        cb(null, data);
+      });
+    };
+
+    this._loadItems(callback, retrieveKeys);
+  }
```

Comment:
You are right here, I believe i was missing the part that the key is already channel:eventId, so instead of re-getting the keys i can just simply map the event-ids directly to keys. So no need for zrange and async aswell.